iothub/device/src/TelemetryMethods.cs

```diff
@@ -29,8 +33,77 @@ public static string GetSqmMachineId()
                     if (Logging.IsEnabled) Logging.Error(null, ex, nameof(TelemetryMethods));
                 }
             }
+            else
+            {
+                try
+                {
+                    // get HMAC-SHA256 of the ADHS application ID, keyed by the machine ID
+                    using var sr = new StreamReader("/etc/machine-id");
```

Comment:
What would this resolve to on a non-linux machine?

Reply:
perhaps consider adding check that this else block is valid only for linux ?

Reply:
Added explicit check for Linux.